src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs

```diff
@@ -67,28 +67,42 @@ public int Run(ParseResult parseResult)
                 });
             }
 
+            _args = [.. parseResult.UnmatchedTokens];
             VSTestTrace.SafeWriteTrace(() => $"Wait for connection(s) on pipe = {_pipeNameDescription.Name}");
             _namedPipeConnectionLoop = Task.Run(async () => await WaitConnectionAsync(_cancellationToken.Token));
 
-            bool containsNoBuild = parseResult.UnmatchedTokens.Any(token => token == CliConstants.NoBuildOptionKey);
-            List<string> msbuildCommandlineArgs = [$"-t:{(containsNoBuild ? string.Empty : "Build;")}_GetTestsProject",
-                 $"-p:GetTestsProjectPipeName={_pipeNameDescription.Name}",
-                    "-verbosity:q"];
-
-            AddAdditionalMSBuildParameters(parseResult, msbuildCommandlineArgs);
-
-            if (VSTestTrace.TraceEnabled)
+            if (parseResult.HasOption(TestCommandParser.TestModules))
```

Comment:
I would refactor the body of the method inside something like `RunWithTestModulesFilter` to keep the entry point short and clear.